nimrod/output_generation/output_generator.py

```diff
@@ -27,9 +28,27 @@ def write_report(self, context: OutputGeneratorContext) -> None:
         file_path = path.join(self.REPORTS_DIRECTORY, self._report_name)
 
         logging.info(f"Starting data processing of {self._report_name} report")
-        data = self._generate_report_data(context)
+        new_data = self._generate_report_data(context)
         logging.info(f"Finished data processing of {self._report_name} report")
 
-        with open(file_path, "w") as write:
-            json.dump(data, write)
+        existing_data = self._load_existing_data(file_path)
+
+        if not isinstance(existing_data, list):
+            existing_data = [existing_data] if existing_data else []
+        existing_data.append(new_data)
+
+        self._write_json(file_path, existing_data)
         logging.info(f"Finished generation of {self._report_name} report")
+
+    def _load_existing_data(self, file_path: str):
+        if not path.exists(file_path):
+            return []
+        try:
+            with open(file_path, "r") as read_file:
+                return json.load(read_file)
+        except json.JSONDecodeError:
+            return []
+
+    def _write_json(self, file_path: str, data) -> None:
+        with open(file_path, "w") as write_file:
+            json.dump(data, write_file, indent=4)
```

Comment:
A ideia aqui é abrir dados existentes de uma execução anterior e continuar a escrever neles, certo?

Acho que vale um comentário explicando a intenção do _load_existing_data. Algo como

`Loads data stored from previous runs so new data is appended instead of overwriting`

Uma ideia talvez seja possibilitar passar uma flag pra escolher se queremos "zerar" tudo ou continuar escrevendo no arquivo que existe. Mas não precisa alterar isso nesse PR, é apenas uma ideia. Também não sei o quão útil isso seria.